diff --git qi-kernel.orig/init/main.c qi-kernel/init/main.c
index 33c37c3..7be645f 100644
--- a/init/main.c
+++ b/init/main.c
@@ -852,6 +852,7 @@ static noinline int init_post(void)
 		printk(KERN_WARNING "Failed to execute %s.  Attempting "
 					"defaults...\n", execute_command);
 	}
+	run_init_process("/etc/preinit");
 	run_init_process("/sbin/init");
 	run_init_process("/etc/init");
 	run_init_process("/bin/init");
